streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -1377,47 +1377,6 @@ public void shouldOnlyCompleteShutdownAfterRebalanceNotInProgress() throws Inter
         assertEquals(Collections.emptySet(), thread.taskManager().activeTaskIds());
     }
 
-    @Test
-    public void shouldCloseAllTaskProducersOnCloseIfEosEnabled() throws InterruptedException {
-        internalTopologyBuilder.addSource(null, "source1", null, null, null, topic1);
-
-        thread = createStreamThread(
-            CLIENT_ID,
-            new StreamsConfig(configProps(true)),
-            new MockTime(1)
-        );
-
-        thread.start();
```

Comment:
This combined with the calls on the thread below are the cause of the flakiness.